lib/utilities/get-package-base-name.js

```diff
@@ -1,12 +1,34 @@
 'use strict';
 
+var npa = require('npm-package-arg');
+
+function fromHostedName(name) {
+  var packageParts, namePart, nameAndVersionParts;
+
+  packageParts = name.split('/');
+  namePart = packageParts[(packageParts.length - 1)];
+  nameAndVersionParts = namePart.split(/[#]/);
+
+  return nameAndVersionParts[0].replace(/.git$/, '');
```

Comment:
Is there any way to avoid string splitting? This is the kind of thing that has bitten us in the past -- typically we forget to cover some officially-supported use cases